clang/lib/CIR/CodeGen/CIRGenClass.cpp

```diff
@@ -865,39 +865,55 @@ Address CIRGenFunction::getAddressOfBaseClass(
     bool nullCheckValue, SourceLocation loc) {
   assert(!path.empty() && "Base path should not be empty!");
 
+  CastExpr::path_const_iterator start = path.begin();
```

Comment:
oof... there is a lot of 'pointer-offset-dancing' happening here that is not particularly lovely.  I DO wonder if this is begging for an iterator_range though.

Reply:
It's not clear to me what you're suggesting here. Can you elaborate?